docs/datatypes/stream/index.md

```diff
@@ -227,22 +227,22 @@ Just like Streams, sinks are super compositional. Sink's operators allow us to c
 
-### Transducer
+### Pipeline
 
-With `Transducer`s, we can transform streams from one type to another, in a **stateful fashion**, which is sometimes necessary when we are doing encoding and decoding. 
+With `Pipeline`s, we can transform streams from one type to another, in a **stateful fashion**, which is sometimes necessary when we are doing encoding and decoding. 
 
-Transducer is a transformer of element types. Transducer accepts some element of type `A` and produces some element of type `B`, and it may fail along the way or use the environment. It just transforms elements from one type to another type in a stateful way. 
+Pipeline is a transformer of element types. Pipeline accepts some element of type `A` and produces some element of type `B`, and it may fail along the way or use the environment. It just transforms elements from one type to another type in a stateful way. 
 
-For example, we can write counter with transducers. We take strings and then split them into lines, and then we take the lines, and we split them into words, and then we count them. 
+For example, we can write counter with pipelines. We take strings and then split them into lines, and then we take the lines, and we split them into words, and then we count them. 
 
-Another common use case of transducers is **writing codecs**. We can use them to decode the bytes into strings. We have a bunch of bytes, and we want to end up with a JSON and then once we are in JSON land we want to go from JSON to our user-defined data type. So, by writing a transducer we can convert that JSON to our user-defined data type.
+Another common use case of pipelines is **writing codecs**. We can use them to decode the bytes into strings. We have a bunch of bytes, and we want to end up with a JSON and then once we are in JSON land we want to go from JSON to our user-defined data type. So, by writing a pipeline we can convert that JSON to our user-defined data type.
 
-**Transducers are very efficient**. They only exist for efficiency reasons because we can do everything we need actually with Sinks. Transducers exist only to make transformations faster. Sinks are not super fast to change from one sink to another. So transducers were invented to make it possible to transform element types in a compositional way without any of the performance overhead associated with changing over a Sink. 
+**Pipelines are very efficient**. They only exist for efficiency reasons because we can do everything we need actually with Sinks. Pipelines exist only to make transformations faster. Sinks are not super fast to change from one sink to another. So pipelines were invented to make it possible to transform element types in a compositional way without any of the performance overhead associated with changing over a Sink. 
```

Comment:
I don't think this paragraph is true for the pipelines as they are both implemented with channels

Reply:
I think the idea was that we don't have to keep acquiring and releasing resources associated with a sink when we transduce with it but let's just delete it.